src/main/scala/com/spotify/ratatool/samplers/BigSampler.scala

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.ratatool.samplers
+
+import java.io.IOException
+import java.net.URI
+import java.nio.charset.Charset
+
+import com.google.api.client.googleapis.services.AbstractGoogleClientRequest
+import com.google.api.client.util.{BackOff, BackOffUtils, Sleeper}
+import com.google.api.services.bigquery.model.{Table, TableFieldSchema, TableReference, TableSchema}
+import com.google.cloud.dataflow.sdk.io.BigQueryIO
+import com.google.cloud.dataflow.sdk.options.BigQueryOptions
+import com.google.cloud.dataflow.sdk.transforms.SerializableFunction
+import com.google.cloud.dataflow.sdk.util.{FluentBackoff, Transport}
+import com.google.cloud.hadoop.util.ApiErrorExtractor
+import com.google.common.hash.{HashCode, HashFunction, Hasher, Hashing}
+import com.google.protobuf.ByteString
+import com.spotify.ratatool.io.{AvroIO, FileStorage}
+import com.spotify.ratatool.serde.JsonSerDe
+import com.spotify.scio.bigquery.TableRow
+import com.spotify.scio.io.{Tap, Taps}
+import com.spotify.scio.{ContextAndArgs, ScioContext, _}
+import org.apache.avro.Schema
+import org.apache.avro.Schema.{Field, Type}
+import org.apache.avro.generic.GenericRecord
+import org.joda.time.Duration
+import org.slf4j.LoggerFactory
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+import scala.language.existentials
+import scala.util.{Failure, Success, Try}
+
+
+object BigSampler {
+
+  private val log = LoggerFactory.getLogger(BigSampler.getClass)
+  private val utf8Charset = Charset.forName("UTF-8")
+
+  private val kissHashFun = Hashing.murmur3_128(System.currentTimeMillis.toInt)
```

Comment:
Some resources in:
 * http://softwareengineering.stackexchange.com/questions/49550/which-hashing-algorithm-is-best-for-uniqueness-and-speed
 * http://stackoverflow.com/questions/11899616/murmurhash-what-is-it

afaiu there won't be guarantee, but it should be reasonably good distribution on different types while still performant. Also the law of big numbers should kick in as well, that said there might be corner cases (highly skewed data) where this will get more data then specified, that said i think that is fine, the `pct` is "approximate" and I believe it's fine that way.